kzg/src/polynomial.rs

```diff
@@ -0,0 +1,82 @@
+use pairing_ce::{
+    bls12_381::{Fr, G1Affine},
+    ff::Field,
+    GenericCurveAffine, GenericCurveProjective,
+};
+use rand::{OsRng, Rng};
+
+pub struct Polynomial {
+    pub coeffs: Vec<Fr>,
+}
+
+impl Polynomial {
+    pub fn new(coeffs: Vec<Fr>) -> Self {
+        Polynomial { coeffs }
+    }
+
+    pub fn rand(degree: usize) -> Self {
+        let mut rng = OsRng::new().unwrap();
+        let coeffs = (0..degree).map(|_| rng.gen::<Fr>()).collect();
+        Polynomial { coeffs }
+    }
+
+    pub fn evaluate(&self, x: Fr) -> Fr {
```

Comment:
if you don't need to take ownership of `x`, make `x` a reference type
```suggestion
    pub fn evaluate(&self, x: &Fr) -> Fr {
```